server/api/src/main/scala/com/prisma/api/database/DatabaseMutationBuilder.scala

```diff
@@ -287,6 +295,20 @@ object DatabaseMutationBuilder {
     } yield action
   }
 
+  def ifThenElseNestedUpsert(condition: SqlStreamingAction[Vector[Boolean], Boolean, Effect],
+                             thenMutactions: DBIOAction[Unit, NoStream, Effect],
+                             elseMutactions: DBIOAction[Unit, NoStream, Effect],
+                             elseChecks: NestedCreateRelationMutaction) = {
+    import scala.concurrent.ExecutionContext.Implicits.global
+    for {
+      exists <- condition
+      action <- if (exists.head) { thenMutactions } else {
+                 val list = List(elseMutactions) ++ elseChecks.requiredCheck ++ elseChecks.removalActions ++ elseChecks.addAction
```

Comment:
the order of checks here is the same as in `.execute()` in the Mutaction. It might be a good idea to refactor those actions in a separate val, e.g.: `elseChecks.allActions`

Reply:
I have problems understanding the name `elseChecks`. It is also acting on the relation data itself. Should it maybe be named `relationMutactions`?

Reply:
Done